helm_drydock/drivers/node/maasdriver/driver.py

```diff
@@ -0,0 +1,306 @@
+                    
+                    if len(matching_fabrics) > 0:
+                        # Fabric exists, update VLAN
+                        fabric = matching_fabrics[0]
+
+                        vlan_list = maas_vlan.Vlans(self.maas_client, fabric_id=fabric.resource_id)
+                        vlan_list.refresh()
+                        vlan_id = n.vlan_id if n.vlan_id is not None else 0
+                        matching_vlans = vlan_list.query({'vid': vlan_id})
+
+                        if len(matching_vlans) > 0:
+                            vlan = matching_vlans[0]
+
+                            vlan.name = n.name
+                            if getattr(n, 'mtu', None) is not None:
+                                vlan.mtu = n.mtu
+
+                            if subnet is not None:
+                                subnet.vlan = vlan.resource_id
+                                subnet.update()
+                            vlan.update()
+                        else:
+                            vlan = maas_vlan.Vlan(self.maas_client, name=n.name, vid=vlan_id,
+                                                mtu=getattr(n, 'mtu', None),fabric_id=fabric.resource_id)
+                            vlan = vlan_list.add(vlan)
+
+                            if subnet is not None:
+                                subnet.vlan = vlan.resource_id
+                                subnet.update()
+
+                    else:
+                        new_fabric = maas_fabric.Fabric(self.maas_client, name=n.name)
+                        new_fabric = fabric_list.add(new_fabric)
+                        new_fabric.refresh()
+                        fabric = new_fabric
+
+                        vlan_list = maas_vlan.Vlans(self.maas_client, fabric_id=new_fabric.resource_id)
+                        vlan_list.refresh()
+                        vlan = vlan_list.single()
+
+                        vlan.name = n.name
+                        vlan.vid = n.vlan_id if n.vlan_id is not None else 0
+                        if getattr(n, 'mtu', None) is not None:
+                            vlan.mtu = n.mtu
+
+                        vlan.update()
+
+                        if subnet is not None:
+                            subnet.vlan = vlan.resource_id
+                            subnet.update()
+
+                    if subnet is None:
+                        subnet = maas_subnet.Subnet(self.maas_client, name=n.name, cidr=n.cidr, fabric=fabric.resource_id,
+                                                vlan=vlan.resource_id, gateway_ip=n.get_default_gateway())
+
+                        subnet_list = maas_subnet.Subnets(self.maas_client)
+                        subnet = subnet_list.add(subnet)
+
+            subnet_list = maas_subnet.Subnets(self.maas_client)
+            subnet_list.refresh()
+
+            action_result = hd_fields.ActionResult.Incomplete
+
+            success_rate = 0
+
+            for n in design_networks:
```

Comment:
I took the tact of updating and then validating, but if the performance overhead is too much validation can be incorporated into the updates.